### DISCUSSION TITLE
max_input_var hit a lot of time

### DISCUSSION THREAD

**Discussion by lenvanessen** [craftcms/cms#8521] [Ideas]
Hi,

You can pretty easily hit your `max_input_var` when handling more complex matrix fields.
Could it be an idea to serialize the post request, so we don't have to update all environments to handle updating these fields?

**Comment by jlawrence-yellostudio** [comment]
I agree this does tend to happen a lot on sites that have a lot of blocks, we always have to consider what the impact of adding a new field to a block type will do with regards to max input vars as the impact will be multiplied by the number of associated blocks. However I believe that serialising the data will only move the issue over to max_post_size instead.

Often however we've discovered that in the cases where we have lots of blocks (50+) on a single entry, we should have really done them as many entries with a single matrix block within each (which benefits by not filling the content table with fields specific for that entry type) and then used some categories to group these entries together.

This resolves much of the issues with CP performance (loading a page with 100+ matrix blocks can take 30+ seconds) but admittedly doing this as an afterthought is quite a refactor.
